src/main.cpp

```diff
@@ -1,6 +1,22 @@
 #include <iostream>
 
 int main(int argc, char *argv[]){
-    std::cout << "Hello,World!" << std::endl;
+    // set a limit of the number of arguments to be passed: in argc
+    if(argc > 4)
+    {
+        // tell the user how to run the program
+        std::cout<<"Usage1: passgen -L length_of_the_password_you_want\n";
+        // We can add other usage messages here based on the input error committed by the user.
+        return 1;
+    }
+
+    // set the length of the character set
+    if(argv[1] == "-L")
+    {
+        int length = atoi(argv[2]);
+        // generate length using length generator.
```

Comment:
There are a couple of problems with this approach to parse. Your method assumes that `-L` option is present in `argv[1]` which may not always be the case as we also have a `-S` option which may come before `-L`. This needs rewrite

Reply:
Yes, just changed the code. And I think, wouldn't it be feasible to design a specific format for command-line arguments, for example," passgen -S -L 20". That is we can add this instruction in how to use the generator in the "-H' section.